src/screens/TapsScreen.js

```diff
@@ -0,0 +1,23 @@
+// @flow
+
+import * as React from 'react';
+import HeaderIcon from '../common/HeaderIcon';
+import TapsList from '../components/TapsList';
+
+class TapsScreen extends React.Component<Props> {
+  static navigationOptions = ({ navigation }: Object): Object => ({
```

Comment:
Is this part of the router? This seems like a really weird pattern.

Reply:
yes, it is.
https://reactnavigation.org/docs/navigators/navigation-options
And yes, it looks weird and pretty stupid for me, also..It completely breaks react-component lifecycle conventions.